src/test/java/org/apache/datasketches/CrossCheckQuantilesTest.java

```diff
@@ -0,0 +1,356 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.datasketches;
+
+import static org.apache.datasketches.CrossCheckQuantilesTest.PrimType.DOUBLE;
+import static org.apache.datasketches.CrossCheckQuantilesTest.PrimType.FLOAT;
+import static org.apache.datasketches.CrossCheckQuantilesTest.SkType.CLASSIC;
+import static org.apache.datasketches.CrossCheckQuantilesTest.SkType.KLL;
+import static org.apache.datasketches.CrossCheckQuantilesTest.SkType.REQ;
+import static org.apache.datasketches.CrossCheckQuantilesTest.SkType.REQ_NO_DEDUP;
+import static org.apache.datasketches.CrossCheckQuantilesTest.SkType.REQ_SV;
+import static org.testng.Assert.assertEquals;
+
+import org.apache.datasketches.kll.KllFloatsSketch;
+import org.apache.datasketches.quantiles.DoublesSketch;
+import org.apache.datasketches.quantiles.UpdateDoublesSketch;
+import org.apache.datasketches.req.ReqSketch;
+import org.apache.datasketches.req.ReqSketchBuilder;
+import org.apache.datasketches.req.ReqSketchSortedView;
+import org.testng.annotations.Test;
+
+public class CrossCheckQuantilesTest {
+
+  enum SkType { REQ, REQ_SV, REQ_NO_DEDUP, KLL, CLASSIC }
+
+  enum PrimType { DOUBLE, FLOAT }
+
+  final int k = 32; //all sketches are in exact mode
+  final boolean hra = false; //for the REQ sketch
+  final boolean INCLUSIVE = true;
+  final boolean NON_INCLUSIVE = false;
+
+  @Test
+  public void checkQuantileSketches() {
+    checkQAndR(REQ, FLOAT, NON_INCLUSIVE); //must do REQ first to compute expected results!
+    checkQAndR(REQ, FLOAT, INCLUSIVE);
+
+    checkQAndR(REQ_SV, FLOAT, NON_INCLUSIVE);
+    checkQAndR(REQ_SV, FLOAT, INCLUSIVE);
+
+    checkQAndR(REQ_NO_DEDUP, FLOAT, NON_INCLUSIVE);
+    checkQAndR(REQ_NO_DEDUP, FLOAT, INCLUSIVE);
+
+    checkQAndR(KLL, FLOAT, NON_INCLUSIVE);
+    checkQAndR(KLL, FLOAT, INCLUSIVE);
+
+    checkQAndR(CLASSIC, DOUBLE, NON_INCLUSIVE);
+    checkQAndR(CLASSIC, DOUBLE, INCLUSIVE);
+    println("");
+  }
+
+  double[] testRankResults_NI = null;
+  double[] testRankResults_I = null;
+  float[] testQuantileFResults_NI = null;
+  float[] testQuantileFResults_I = null;
+  double[] testQuantileDResults_NI = null;
+  double[] testQuantileDResults_I = null;
+
+  private void checkQAndR(final SkType skType, final PrimType type, final boolean inclusive) {
+    String head = "CHECK " + skType.toString();
+    if (inclusive) { println("\n---------- " + head + " INCLUSIVE ----------\n"); }
+    else { println("\n########## " + head + " NON-INCLUSIVE ##########\n"); }
+
+    //CREATE EMPTY SKETCHES
+    ReqSketchBuilder reqBldr = ReqSketch.builder();
+    reqBldr.setK(4).setHighRankAccuracy(hra).setLessThanOrEqual(inclusive);
+    ReqSketch reqSk = reqBldr.build();
+
+    KllFloatsSketch kllSk = KllFloatsSketch.newHeapInstance(k);
+
+    UpdateDoublesSketch udSk = DoublesSketch.builder().setK(k).build();
```

Comment:
Why is k set to 4 for the REQ sketch, but the constant k=32 used for the classic and KLL sketches? Is the former interpreted as lgK?

Reply:
Oversight.  I can be 32.  No, it is not LgK.